easybuild/easyblocks/generic/gopackage.py

```diff
@@ -0,0 +1,53 @@
+# -*- coding: utf-8 -*-
+##
+# Copyright 2009-2019 Ghent University
+#
+# This file is part of EasyBuild,
+# originally created by the HPC team of Ghent University (http://ugent.be/hpc/en),
+# with support of Ghent University (http://ugent.be/hpc),
+# the Flemish Supercomputer Centre (VSC) (https://www.vscentrum.be),
+# Flemish Research Foundation (FWO) (http://www.fwo.be/en)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# https://github.com/easybuilders/easybuild
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+EasyBuild support for building and installing Go packages, implemented as an easyblock
+
+@author: Bob Dröge (University of Groningen)
+"""
+from easybuild.easyblocks.generic.tarball import Tarball
+from easybuild.tools.build_log import EasyBuildError
+from easybuild.tools.run import run_cmd
+from easybuild.tools.modules import get_software_root
+
+
+class GoPackage(Tarball):
+    """
+    Install a Go package as a separate module, or as an extension.
+    """
+
+    def install_step(self):
```

Comment:
Since go compiles the packages, wouldn't it make sense to do this in the `build_step`?

Reply:
I think I did this because the install_step somehow removed everything that was written to installdir by build_step, even if I override the install_step and let it do nothing. Is there a way to prevent that?

Reply:
I don't think the `install_step` does that if you derive from `EasyBlock`. But the cleanup might. Take a look at the `buildininstalldir` option, that might be what you need.

Reply:
It still seems to do this when deriving from EasyBlock. Adding buildinstalldir=True puts all the source files in the installdir, which is not what we want. The issue seems to be that the install_step cleans the installdir (because it might be an old installation), and hence removes the executable that the build_step installed. I guess we can prevent this with keeppreviousinstall=True, but I don't think that's a good idea. Any other ideas?